### PR TITLE
Add `this` to the locals viewer

### PR DESCRIPTION
**Pull request by TatriX**

### CODE REVIEW

File: indium-debugger.el [hunk 1]
```diff
@@ -351,7 +351,20 @@ Return nil if no local file can be found."
 
 (defun indium-debugger-get-current-scopes ()
   "Return the scope of the current stack frame."
-  (map-elt (indium-debugger-current-frame) 'scope-chain))
+  (seq-sort-by
+   (lambda (scope)
+     ;; Possible protocol values: global, local, with, closure, catch, block, script, eval, module.
+     ;; We also add "this" to the list.
+     (pcase (map-elt scope 'type)
```
Comment (NicolasPetton) [inline indium-debugger.el]: That's too specific to Chrome's debugger protocol to be in `indium-debugger.el`.

Comment (TatriX) [inline indium-debugger.el]: Should I extract sorting function to a generic method or?

**Comment by NicolasPetton** [comment]
Why does it need sorting if it didn't before?

**Comment by TatriX** [comment]
I'd like to see usefull scopes first.
Local variables, then `this` context, then script variables.
As I append `this` and there can be several block scopes I added sorting.

**Comment by NicolasPetton** [comment]
TatriX <notifications@github.com> writes:

> I'd like to see usefull scopes first.
> Local variables, then `this` context, then script variables.
> As I append `this` and there can be several block scopes I added
> sorting.

AFAIR, the order of the scopes today are the ones of the scope chain, so
it makes sense to keep it this way.

**Comment by TatriX** [comment]
So I should determine where to insert `this` scope?

**Comment by NicolasPetton** [comment]
TatriX <notifications@github.com> writes:

> So I should determine where to insert `this` scope?

I guess so :)

Can you check how and where Chrome displays it?

**Comment by TatriX** [comment]
Chrome adds `this` as a last element of the `Local` subtree.

**Comment by NicolasPetton** [comment]
Thanks!

**Pull request merged**